src/markdown-pages/contribute-to-quickstarts/instrument-library/send-events.mdx

```diff
@@ -0,0 +1,170 @@
+-  [Events at New Relic](https://docs.newrelic.com/docs/data-apis/understand-data/new-relic-data-types/#events-new-relic)
+-  [Formatting JSON](https://docs.newrelic.com/docs/data-apis/ingest-apis/introduction-event-api#instrument)
+
+## Send your events to New Relic
+There are a few ways we will discuss in order to send your event data into New Relic. We have our [telemetry SDK's](https://docs.newrelic.com/docs/data-apis/ingest-apis/telemetry-sdks-report-custom-telemetry-data), which are development kits instrumented to help you easily set up a POST request via several very common languages (C, Java, Python, .NET, etc.). We also have a basic curl command, which can be used on a windows or linux system to make a POST request to New Relic. We also have a prometheus option, if you use prometheus and have difficulty with the first two options.
+
+### New Relic License Key
+
+You will need to set up an environment variable for your users to input their [New Relic License key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#ingest-license-key), or give an option for them to add it to the request header. The examples below will assume this variable is configured as `NEW_RELIC_INSERT_KEY`.
+
+### Send events via a curl/PowerShell command
+
+First, let's cover how you can send agentless data to New Relic, with just a simple command. These commands can be used in Linux or Windows environments. Here are some example requests for both
+
+* Linux/Bash
+
+```bash
+gzip -c example_events.json | curl -X POST -H "Content-Type: application/json" -H "Api-Key: NEW_RELIC_INSERT_KEY" -H "Content-Encoding: gzip" https://insights-collector.newrelic.com/v1/accounts/events --data-binary @-
+```
+* Windows/PowerShell
+```bash
+
+# Replace with your custom event for the body
+$body = $example_events
+
+$headers = @{}
+$headers.Add("Api-Key", "$NEW_RELIC_INSERT_KEY")
+$headers.Add("Content-Encoding", "gzip")
+
+
+$encoding = [System.Text.Encoding]::UTF8
+$enc_data = $encoding.GetBytes($body)
+
+$output = [System.IO.MemoryStream]::new()
+$gzipStream = New-Object System.IO.Compression.GzipStream $output, ([IO.Compression.CompressionMode]::Compress)
+
+$gzipStream.Write($enc_data, 0, $enc_data.Length)
+$gzipStream.Close()
+$gzipBody = $output.ToArray()
+
+Invoke-WebRequest -Headers $headers -Method Post -Body $gzipBody  "https://insights-collector.newrelic.com/v1/accounts/events"
+```
+
+
+
+### Using our SDK's
+
+New Relic has a set of open source API client libraries that send data to New Relic. These operate by using our API's for data ingest, and in this case our event API. We have two SDK's that work with events, and those are Python and Java. Here we will walk through how to install each, and give a code example using both SDK's.
+
+For Python, you'll need to install the `newrelic-telemetry-sdk` package. you can do this by using pip:
+
+```bash
+$ pip install newrelic-telemetry-sdk
+```
+For Java, you will need to set up several dependencies:
+```java
```

Comment:
Is there anything else the user has to do to install the dependencies? Can you provide steps for that? Also, have you tried out all this code to make sure it works?

Reply:
Just updating on this. They don't need to do anything other than installing and running gradle build, so I actually merged the decencies with the code block and just mentioned running gradle. I have run the Python code and it works, I am still trying to set up java and the dependencies to run the java snip but it's a direct example from our [example events docs](https://github.com/newrelic/newrelic-telemetry-sdk-java/blob/main/telemetry_examples/src/main/java/com/newrelic/telemetry/examples/EventExample.java) so it should work.